TeaDiary/Scripts/App/ApiCommands.js

```diff
@@ -1,29 +1,43 @@
 ﻿function ApiCommands() {
     var self = this;
 
-    self.addTea = (tea, callback) => {
+    self.addTea = (tea, successCallback, errorCallback) => {
         $.ajax("/api/tea/",
             {
                 method: 'POST',
                 data: tea
             }
-        ).done((response) => { callback(response) });
+        ).done((response) => {
+            successCallback(response);
+        });
```

Comment:
Alongside with `.done()` there is `.error()` (or something like that) which will trigger if there is an error (HTTP level error or HTTP error code returned from the server)